syntax = "proto2";

package oeo;

message VehicleInformation {
    required uint32 id = 1;
    required VehicleInformationData data = 2;
}

message VehicleInformationData {

    required Status status = 1;
    required float battery_voltage = 2;
    required float battery_current = 3;
    required float battery_percent = 4;
    required float latitude = 5;
    required float longitude = 6;
    required float altitude = 7;
    required Velocity velocity = 8;
}

enum Status {
    DISARMED = 0;
    ARMED = 1;
}

enum Mode {
    DRONE_GUIDED    = 1;
    DRONE_ALT_HOLD  = 2;
    DRONE_MISC      = 4;
    
    ROVER_AUTO      = 8;
    ROVER_MANUAL    = 16;
    ROVER_MISC      = 32;
}

message Velocity {
    required float north = 1;
    required float east = 2;
    required float down = 3;
}